plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +50,24 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
```

Comment:
There is no concurrent access in the code you have today.  If you are expecting concurrent access, you are broken in other ways.  Can you explain how you expect parallelism to occur here?